hera_cal/lst_stack/calibration.py

```diff
@@ -132,240 +174,350 @@ def lstbin_absolute_calibration(
+    reference model. Each day is calibrated independently using only the absolute calibration
+    degrees of freedom (per-antenna calibration is not currently supported). The calibration
+    is performed in two steps:
 
-            # Store gains to return
-            if return_gains:
-                gains[(ant1, ant2, pol)] = bl_gain
+        1. Amplitude calibration: The amplitude of the data is scaled to match the model using
+           hera_cal.abscal.abs_amp_lincal to perform gain fits.
+        2. Phase calibration: The phase of the data is adjusted to match the model by fitting
+           for the tip-tilt abscal degeneracy using hera_cal.abscal.complex_phase_abscal. Note
+           that while the tip-tilt degeneracy may be solved for in greater than 2 dimensions
+           in the general case, this function only solves for the 2D tip-tilt degeneracy.
 
-    if auto_stack:
-        for polidx, pol in enumerate(pols):
-            for apidx, (ant1, ant2) in enumerate(auto_stack.antpairs):
-                bl_gain = np.ones(
-                    (stack.nights.size, stack.freq_array.size), dtype=complex
-                )
-                for bandidx, band in enumerate(inpaint_bands):
-                    # Construct the raw gain and remove nans and infs
-                    raw_gain = calibration_parameters[f"A_J{pol}"] ** 2
-                    raw_gain = np.where(
-                        stack.flags[:, 0, band, polidx], 1.0 + 0.0j, raw_gain[:, band]
-                    )
 
-                    # Compute smooth gain for each parameter and remove zeros/nans/infs
-
-                    if smooth_gains:
-                        basis = smoothing_functions[bandidx]
-                        bl_gain_here = np.array(
-                            [
-                                np.dot(basis, _fmat.dot(_raw_gain))
-                                for _fmat, _raw_gain in zip(
-                                    fmats[pol][bandidx], raw_gain
-                                )
-                            ]
-                        )
-                    else:
-                        bl_gain_here = raw_gain
-
-                    bl_gain_here = np.where(
-                        np.isfinite(bl_gain_here), bl_gain_here, 1.0 + 0.0j
-                    )
-                    bl_gain_here = np.where(
-                        np.isclose(bl_gain_here, 0), 1.0 + 0.0j, bl_gain_here
-                    )
-                    bl_gain[:, band] = bl_gain_here
+    Parameters:
+    ----------
+        stack : LSTStack
+            The LSTStack object to calibrate
+        model : np.ndarray
+            The reference model to calibrate the data to. The model should have the same
+            number of baselines, frequencies, and polarizations as the data in stack. The model
+            can simply be the mean of the data in the stack if data are already abscal'd.
```

Comment:
I'd clarify which axis you suggest taking the mean over. It's the 0th, right?

Reply:
Good catch, I've clarified that here